src/field/p256_scalar.rs

```diff
@@ -0,0 +1,271 @@
+        )
+    }
+
+    #[inline]
+    fn from_canonical_u64(n: u64) -> Self {
+        Self([n, 0, 0, 0])
+    }
+    #[inline]
+    fn from_noncanonical_u128(n: u128) -> Self {
+        Self([n as u64, (n >> 64) as u64, 0, 0])
+    }
+    #[inline]
+    fn from_noncanonical_u96(n: (u64, u32)) -> Self {
+        Self([n.0, n.1 as u64, 0, 0])
+    }
+
+    fn from_noncanonical_i64(n: i64) -> Self {
+        let f = Self::from_canonical_u64(n.unsigned_abs());
+        if n < 0 {
+            -f
+        } else {
+            f
+        }
+    }
+    fn from_noncanonical_u64(n: u64) -> Self {
+        Self::from_canonical_u64(n)
+    }
+}
+
+impl PrimeField for P256Scalar {
+    fn to_canonical_biguint(&self) -> BigUint {
+        let mut res = biguint_from_array(self.0);
+        if res >= Self::order() {
+            res -= Self::order();
+        }
+        res
+    }
+}
+
+/* -------------------------------------------------------------------------- */
+/*                             arithmetic impls                               */
+/* -------------------------------------------------------------------------- */
+
+impl Neg for P256Scalar {
+    type Output = Self;
+    #[inline]
+    fn neg(self) -> Self {
+        if self.is_zero() {
+            Self::ZERO
+        } else {
+            Self::from_noncanonical_biguint(Self::order() - self.to_canonical_biguint())
+        }
+    }
+}
+impl Add for P256Scalar {
+    type Output = Self;
+    #[inline]
+    fn add(self, rhs: Self) -> Self {
+        let mut res = self.to_canonical_biguint() + rhs.to_canonical_biguint();
+        if res >= Self::order() {
+            res -= Self::order();
+        }
+        Self::from_noncanonical_biguint(res)
+    }
+}
+impl AddAssign for P256Scalar {
+    #[inline]
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+impl Sum for P256Scalar {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::ZERO, |acc, x| acc + x)
+    }
+}
+
+impl Sub for P256Scalar {
+    type Output = Self;
+    #[inline]
+    fn sub(self, rhs: Self) -> Self {
+        self + -rhs
+    }
+}
+impl SubAssign for P256Scalar {
+    #[inline]
+    fn sub_assign(&mut self, rhs: Self) {
+        *self = *self - rhs;
+    }
+}
+
+impl Mul for P256Scalar {
+    type Output = Self;
+    #[inline]
+    fn mul(self, rhs: Self) -> Self {
+        Self::from_noncanonical_biguint(
+            (self.to_canonical_biguint() * rhs.to_canonical_biguint()).mod_floor(&Self::order()),
+        )
+    }
+}
+impl MulAssign for P256Scalar {
+    #[inline]
+    fn mul_assign(&mut self, rhs: Self) {
+        *self = *self * rhs;
+    }
+}
+impl Product for P256Scalar {
+    fn product<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.reduce(|acc, x| acc * x).unwrap_or(Self::ONE)
+    }
+}
+
+impl Div for P256Scalar {
+    type Output = Self;
+    fn div(self, rhs: Self) -> Self {
+        self * rhs.inverse()
```

Comment:
This silences Clippy:
```suggestion
        core::iter::once(self).fold(rhs.inverse(), |a, b| a * b)
```